packages/server/modules/serverinvites/services/validation.ts

```diff
@@ -0,0 +1,120 @@
+import { UserRecord } from '@/modules/core/helpers/types'
+import { CreateInviteParams } from '@/modules/serverinvites/domain/operations'
+import { InviteCreateValidationError } from '@/modules/serverinvites/errors'
+import { ResourceTargets, isServerInvite, resolveTarget } from '../helpers/inviteHelper'
+import { UserWithOptionalRole } from '@/modules/core/repositories/users'
+import { authorizeResolver } from '@/modules/shared'
+import { Roles } from '@speckle/shared'
+import { getStreamCollaborator } from '@/modules/core/repositories/streams'
+import { TokenResourceIdentifier } from '@/modules/serverinvites/services/operations'
+
+/**
+ * Validate invite creation input data
+ */
+export async function validateInput(
```

Comment:
Shouldn't these be factories? On the one hand they're not that complicated and essentially helpers, but on the other - there are real dependencies, e.g. authorizeResolver and getStreamCollaborator